src/Ide/Types.hs

```diff
@@ -57,8 +60,24 @@ data PluginDescriptor =
                    , pluginSymbolsProvider    :: !(Maybe SymbolsProvider)
                    , pluginFormattingProvider :: !(Maybe (FormattingProvider IO))
                    , pluginCompletionProvider :: !(Maybe CompletionProvider)
+                   , pluginRenameProvider     :: !(Maybe RenameProvider)
                    }
 
+defaultPluginDescriptor :: PluginDescriptor
```

Comment:
I think it would be better to make the `defaultPluginDescriptor` take the `pluginId` as a parameter, rather than initializing it to "".  This fits in easily with the current usage, and setting the plugin id is critical for routing execute command calls. Doing that makes sure it is not accidentally left as "".

Reply:
done